From 48ebe8ec1d732e1ec379336eb5bffe236390a8a5 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Tue, 28 Mar 2023 11:03:40 +0530
Subject: [PATCH] Adding 3 virtual functions added in Android U

3 virtual functions have been added in Android U
health HAL. So the definition of the same is added
in this patch.
If utilized then proper implementation for the same
need to be done.

Tracked-On: NA
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 health/aidl/Health.cpp                   | 12 ++++++++++++
 health/aidl/include/health-impl/Health.h |  3 +++
 2 files changed, 15 insertions(+)

diff --git a/health/aidl/Health.cpp b/health/aidl/Health.cpp
index 3315154..e0242ed 100644
--- a/health/aidl/Health.cpp
+++ b/health/aidl/Health.cpp
@@ -127,6 +127,18 @@ ndk::ScopedAStatus Health::getChargeStatus(BatteryStatus* out) {
                        BatteryStatus::UNKNOWN, out);
 }
 
+ndk::ScopedAStatus Health::setChargingPolicy(BatteryChargingPolicy out) {
+    return ndk::ScopedAStatus::fromExceptionCode(EX_UNSUPPORTED_OPERATION);
+}
+
+ndk::ScopedAStatus Health::getChargingPolicy(BatteryChargingPolicy* out) {
+    return ndk::ScopedAStatus::fromExceptionCode(EX_UNSUPPORTED_OPERATION);
+}
+
+ndk::ScopedAStatus Health::getBatteryHealthData(BatteryHealthData* out) {
+    return ndk::ScopedAStatus::fromExceptionCode(EX_UNSUPPORTED_OPERATION);
+}
+
 ndk::ScopedAStatus Health::getDiskStats(std::vector<DiskStats>*) {
     // This implementation does not support DiskStats. An implementation may extend this
     // class and override this function to support disk stats.
diff --git a/health/aidl/include/health-impl/Health.h b/health/aidl/include/health-impl/Health.h
index 6bd4946..e3770e9 100644
--- a/health/aidl/include/health-impl/Health.h
+++ b/health/aidl/include/health-impl/Health.h
@@ -61,6 +61,9 @@ class Health : public BnHealth, public HalHealthLoopCallback {
     ndk::ScopedAStatus getCurrentAverageMicroamps(int32_t* out) override;
     ndk::ScopedAStatus getCapacity(int32_t* out) override;
     ndk::ScopedAStatus getChargeStatus(BatteryStatus* out) override;
+    ndk::ScopedAStatus setChargingPolicy(BatteryChargingPolicy out) override;
+    ndk::ScopedAStatus getChargingPolicy(BatteryChargingPolicy* out) override;
+    ndk::ScopedAStatus getBatteryHealthData(BatteryHealthData* out) override;
 
     // A subclass may either override these or provide function pointers in
     // in healthd_config in the constructor.
-- 
2.17.1

